### PR TITLE
tweaks

### PR DESCRIPTION
**Pull request by hjanetzek**
Still work in progress
- compute the range of overlapping cells in broadphase instead of checking overlap with each cell
- reuse gridAABBs and set<pair> 
- use templates instead Vec2 to support glm::vec2 directly

### CODE REVIEW

**Comment by tallytalwar** [comment]
Reviewing now.

**Comment by tallytalwar** [comment]
Did you benchmark to see whats the speedup?

**Comment by hjanetzek** [comment]
A bit more than twice as fast in my tests. If you run the test it uses both versions grid1 vs grid2 in the output. The old version should be removed when ready

**Comment by tallytalwar** [comment]
Though this is good for current implementation.

The aim of isect2d was to provide multiple collision detection algos and for the developer using the library to pick one based on his/her convenience. I also agree previously also this was not apparent from the code. Both me and @karimnaaji, want this library to be more generic and not specific to one specific intersection test implementation.

That being said, I am fine with the current specific implementation as its improves on the previous, but we should try to make this more generic to be used for other purposes and other developers besides tangram.

File: include/isect2d.h [hunk 1]
```diff
@@ -2,79 +2,245 @@
...
         }
     }
-    
+
     for (int i = 0; i < n; ++i) {
         auto& v = gridAABBs[i];
+
         for (size_t j = 0; j < v.size(); ++j) {
             for (size_t k = j + 1; k < v.size(); ++k) {
-                if (v[j].index != v[k].index && v[j].aabb->intersect(*v[k].aabb)) {
```
Comment (tallytalwar) [inline include/isect2d.h]: I do not remember why I added the check for index here ... it seems it was an extra redundant check.

File: tests/main.cpp [hunk 2]
```diff
@@ -8,20 +9,28 @@
 #include <stack>
 
 #include <GLFW/glfw3.h>
+#include "../../tangram-es/core/include/glm/glm/glm.hpp"
```
Comment (matteblair) [inline tests/main.cpp]: GLM can be installed through package managers (brew and apt-get, at least) so we can just treat it as a dependency the same as GLFW and not assume that tangram-es is nearby :)

Comment (hjanetzek) [inline tests/main.cpp]: Yeah. This would not have gotten into master. Talked with Varun about it - for this kind of stuff I miss our Native flow :(

**Comment by matteblair** [comment]
Looks good to me. Pretty sophisticated!

Let's swap out that GLM #include to a system include, clean out the old code, then merge this.

File: tests/main.cpp (hunk 2 above)
Comment (tallytalwar) [inline tests/main.cpp]: @hjanetzek we can also use graphics flow for native discussions. Thats what Brett meant when he recommended not having a separate flow for native.

**Pull request merged**